From 4a48421f1be7a2d1230adfb9ff8b848223f54680 Mon Sep 17 00:00:00 2001
From: Felipe Correa da Silva Sanches <juca@members.fsf.org>
Date: Sat, 14 Mar 2026 11:31:19 +0000
Subject: [PATCH 05/25] kn5000: Wire MIDI output (TX0) to midi_port device

Connect Serial Channel 0 TX to a midiout_slot, enabling MIDI output
from the emulated KN5000. Works in conjunction with the UART mode
implementation in the TMP94C241 serial device.
---
 src/mame/matsushita/kn5000.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/mame/matsushita/kn5000.cpp b/src/mame/matsushita/kn5000.cpp
index 993a9031586..fa5055daec9 100644
--- a/src/mame/matsushita/kn5000.cpp
+++ b/src/mame/matsushita/kn5000.cpp
@@ -624,8 +624,10 @@ void kn5000_state::kn5000(machine_config &config)
 	midiin_slot(mdin);
 	mdin.rxd_handler().set(m_maincpu, FUNC(tmp94c241_device::rxd0));
 
-	// TODO: MIDI output
-	// midiout_slot(MIDI_PORT(config, "mdout"));
+	// TX0 = MTXD (MIDI output)
+	auto &mdout(MIDI_PORT(config, "mdout"));
+	midiout_slot(mdout);
+	m_maincpu->txd0().set("mdout", FUNC(midi_port_device::write_txd));
 
 	// RX1/TX1 = CPDATA, SCLK1 = CPSCK — wired to control panel HLE
 	KN5000_CPANEL(config, m_cpanel);
